src/internal/packager/images/pull.go

```diff
@@ -76,11 +76,16 @@ func Pull(ctx context.Context, cfg PullConfig) (map[transform.Image]ocispec.Mani
 	}
 
 	imagesWithOverride := []imageWithOverride{}
+	// Iterate over all images, marking each one as overridden.
 	for _, img := range cfg.ImageList {
 		overriddenImage := img
-		for k, v := range cfg.RegistryOverrides {
-			if strings.HasPrefix(img.Reference, k) {
-				overriddenImage.Reference = strings.Replace(img.Reference, k, v, 1)
+		for _, v := range cfg.RegistryOverrides {
```

Comment:
If I understand this correctly - this indicates that longest prefix will win _when_ the order has longer prefixes first yes?

My gut is leaning toward removing that constraint and sorting the overrides such that we can guarantee longest match.